### DISCUSSION TITLE
Error loading page - Duplicate entry detected.

### DISCUSSION THREAD

**Discussion by adnanbrkovic** [requarks/wiki#3330] [Error / Bug Report]
**Describe the bug**
When trying to load a page, error occurs - "Error while fetching page. Duplicate entry detected. Reload the page to try again."

**To Reproduce**
1. After installation, at welcome screen, click on Create Homepage button
2. After page successfully created, when navigated to page, error occurs - "Error while fetching page. Duplicate entry detected. Reload the page to try again."

**Expected behavior**
To correctly open created page

**Screenshots**

**Host Info (please complete the following information):**
 - OS: Docker/Kubernetes 1.15
 - Wiki.js version: 2.5.170
 - Database engine: postgres 10

**Additional context**
Localization is used in wiki (Croatian language).
I confirm pages are created, and can be edited from Admin area, but it is unable to access them because of error

**Comment by Dharmesh-Makwana** [comment]
I also face the same issue

**Comment by metaColin** [comment]
*Bump* I just did a fresh install of wikijs and tried to create my first page, and am facing exactly the same issue.

Any fix on this?

**Comment by NGPixel** [comment]
This is most likely because the render failed. Check the logs for any error while saving the page.

**Reply by metaColin** [reply]
Logs show the rendering did indeed fail, but I'm still not clear on how to fix this?

Rebooting the docker instance did not help. Im stuck on the  "Error while fetching page. Duplicate entry detected. Reload the page to try again."

Do I have to manually edit the database or something?

**Reply by NGPixel** [reply]
The issue is with some content in your page. Do you have HTML comments (`<!-- -->`) in your page? If so, remove them and save again.

**Reply by metaColin** [reply]
I had just commented out some HTML when the instance crashed.

But how do I get back to the document? The entire server is locked up and I can only view the page the aforementioned error page.

Do I need to use an SQL editor?!?

**Reply by metaColin** [reply]
I ended up just rolling up a fresh instance.

Is this a known issue? HTML comments will break the wiki?

**Reply by NGPixel** [reply]
Yes and it has been fixed already. You can apply this patch in the meantime:
https://github.com/Requarks/wiki/pull/3823/files

**Comment by gramakri** [comment]
Hit this with the Cloudron package as well and it's quite frequently reproducible.

Wiki.js starts out as normal (this is even with `silly` loglevel which seems to have no effect:

```
2022-06-12T18:00:14.520Z [MASTER] info: Syncing locales with Graph endpoint...
2022-06-12T18:00:14.521Z [MASTER] info: Fetching latest updates from Graph endpoint...
2022-06-12T18:00:14.553Z [MASTER] info: Purging orphaned upload files: [ COMPLETED ]
Loading configuration from /app/code/config.yml... OK
2022-06-12T18:00:14.686Z [MASTER] info: Fetching latest updates from Graph endpoint: [ COMPLETED ]
2022-06-12T18:00:14.710Z [JOB] info: Rebuilding page tree...
2022-06-12T18:00:14.851Z [MASTER] info: Pulled latest locale updates for English from Graph endpoint: [ COMPLETED ]
2022-06-12T18:00:14.854Z [MASTER] info: Syncing locales with Graph endpoint: [ COMPLETED ]
2022-06-12T18:00:15.199Z [JOB] info: Using database driver pg for postgres [ OK ]
2022-06-12T18:00:15.248Z [JOB] info: Rebuilding page tree: [ COMPLETED ]
```

```
# curl http://localhost:3000
....
</script><link type="text/css" rel="stylesheet" href="/_assets/css/app.2dd0e8094c075e7da83f.css"><script type="text/javascript" src="/_assets/js/runtime.js?1654986106"></script><script type="text/javascript" src="/_assets/js/app.js?1654986106"></script></head><body><div class="is-fullscreen" id="root"><div class="app-error"><a href="/"><img src="/_assets/svg/logo-wikijs.svg"></a><strong>Oops, something went wrong...</strong><span>Error while fetching page. Duplicate entry detected. Reload the page to try again.</span></div></div></body></html>root@b0aafe9f-4407-4d96-bd57-e13e39d03a57:/app/code# vi server/models/pages.js
```

Debugging further.

(I guess this is a duplicate of https://github.com/requarks/wiki/discussions/3330)

**Reply by gramakri** [reply]
Inspecting the database, the `render` is empty/null:

```
dbb0aafe9f44074d96bd57e13e39d03a57=> select id,path,render from pages;
 id | path | render 
----+------+--------
  1 | home | 
```

**Reply by gramakri** [reply]
Deleting that page and then re-creating the page makes things work!

```
dbb0aafe9f44074d96bd57e13e39d03a57=> DELETE from pages WHERE id=1;
DELETE 1
```

And now, I can create pages. So, it comes down to _why_ was render null to start with.
